semantics/common/builtin.k

```diff
@@ -15,10 +15,11 @@ module COMMON-BUILTIN
      imports K-REFLECTION
      imports C-TYPING-SYNTAX
 
-     syntax Map ::= #nativeDefinedBuiltins() [function, hook(C_SEMANTICS.nativeFunctions), impure]
+     syntax Map ::= #nativeDefinedBuiltins() [function, hook(C_SEMANTICS.nativeFunctions)]
      rule nativeDefinedBuiltins => #nativeDefinedBuiltins()
 
      // defined as a K rule for if the hook is not defined
-     rule #nativeDefinedBuiltins() => .Map
+     rule [[ #nativeDefinedBuiltins() => .Map ]]
```

Comment:
What's the meaning of the `[[` `]]` here?

Reply:
This says that #nativeDefinedBuiltins() rewrites to .Map if we match the pattern after it (in this case it will always match because it's just matching on `<k> _ </k>`. The reason for this is to make it so that the entire configuration is passed as one of the arguments to the hook.